mirrord/config/src/feature/split_queues.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::HashMap;
+
+use mirrord_analytics::CollectAnalytics;
+use mirrord_config_derive::MirrordConfig;
+use schemars::JsonSchema;
+use serde::{self, Deserialize, Serialize};
+
+use crate::config::source::MirrordConfigSource;
+
+/// Maps name of a `MirrordQueueSplitter` resource to a mapping from a fork name to a filter
+/// definition.
+/// ```json
+/// {
+///   "feature": {
+///     "split_queues": {
+///       "splitter": "whatever-q-splitter"
+///       "queues": {
+///         "first-queue": {
+///           "queue_type": "SQS",
+///           "wows": {
+///             "number_attribute": [
+///               { "greater_than_int": 2 },
+///               { "lesser_than_int": 6 }
+///             ]
+///           },
+///           "coolz": {
+///             "string_attribute": "^very .*"
+///           }
+///         }
+///       }
+///     }
+///   }
+/// }
+/// ```
+#[derive(MirrordConfig, Clone, Debug)]
+#[config(derive = "JsonSchema,Eq,PartialEq")]
+pub struct SplitQueuesConfig {
+    /// The name of the `MirrordQueueSplitter` resource to get queue details from.
+    splitter: String,
```

Comment:
Not sure yet about the architecture - will the cluster always have one splitter? If not, does it make sense for the user app to use more than one splitter?

Reply:
There can be one splitter per deployment (or Argo rollout). The app won't have to specify the splitter, it will be automatically derived from the target. I still haven't updated the code since the meeting where that was decided.